Library/Homebrew/cmd/update.sh

```diff
@@ -347,11 +348,20 @@ EOS
            --header "If-None-Match: \"$UPSTREAM_BRANCH_LOCAL_SHA\"" \
            "https://api.github.com/repos/$UPSTREAM_REPOSITORY/commits/$UPSTREAM_BRANCH")"
         [[ "$UPSTREAM_SHA_HTTP_CODE" = "304" ]] && exit
+      elif [[ -n "$HOMEBREW_UPDATE_PREINSTALL" ]]
+      then
+        exit
```

Comment:
I'm not sure I fully understand the intention of this. Does that mean `brew update --preinstall` is a no-op if the remote URL doesn't start with `https://github.com/`, i.e. we cannot check its status sufficiently quickly?